packages/lesswrong/components/common/HashLink.tsx

```diff
@@ -3,12 +3,20 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 // eslint-disable-next-line no-restricted-imports
-import { Link, NavLink } from 'react-router-dom';
+import { Link, LinkProps } from 'react-router-dom';
+
+type ScrollFunction = ((el: HTMLElement) => void);
+
+export type HashLinkProps = Omit<LinkProps, 'to'> & {
+  scroll?: ScrollFunction,
+  to: LinkProps['to'] | null,
```

Comment:
There are a few places in the codebase where we're passing in the results of e.g. a `DatabasePublicSetting<string | null>`, where `.get()` can return `null` (at least in theory, according to the type system).  That worked fine when `Link` was untyped and wasn't checking that we were always passing in a `string` to `to`.  Doesn't work now!

It now [defaults to a no-op](https://github.com/ForumMagnum/ForumMagnum/pull/5255/files#diff-8574800ad05f8bf5a10350f6bfafe36ed500329b7e2bf62e0e55e554f57502c9R88) if it's ever passed a null, or at least I think that's what it's doing.  Open to suggestions here.